internal/controller/apiextensions/composite/composition_transforms.go

```diff
@@ -301,118 +290,30 @@ func stringTrimTransform(input any, t v1.StringTransformType, trim string) strin
-		return strconv.FormatInt(i.(int64), 10), nil
-	},
-	{From: v1.ConvertTransformTypeInt64, To: v1.ConvertTransformTypeBool, Format: v1.ConvertTransformFormatNone}: func(i any) (any, error) { //nolint:unparam // See note above.
-		return i.(int64) == 1, nil
-	},
-	{From: v1.ConvertTransformTypeInt64, To: v1.ConvertTransformTypeFloat64, Format: v1.ConvertTransformFormatNone}: func(i any) (any, error) { //nolint:unparam // See note above.
-		return float64(i.(int64)), nil
-	},
-
-	{From: v1.ConvertTransformTypeBool, To: v1.ConvertTransformTypeString, Format: v1.ConvertTransformFormatNone}: func(i any) (any, error) { //nolint:unparam // See note above.
-		return strconv.FormatBool(i.(bool)), nil
-	},
-	{From: v1.ConvertTransformTypeBool, To: v1.ConvertTransformTypeInt64, Format: v1.ConvertTransformFormatNone}: func(i any) (any, error) { //nolint:unparam // See note above.
-		if i.(bool) {
-			return int64(1), nil
-		}
-		return int64(0), nil
-	},
-	{From: v1.ConvertTransformTypeBool, To: v1.ConvertTransformTypeFloat64, Format: v1.ConvertTransformFormatNone}: func(i any) (any, error) { //nolint:unparam // See note above.
-		if i.(bool) {
-			return float64(1), nil
-		}
-		return float64(0), nil
-	},
-	{From: v1.ConvertTransformTypeFloat64, To: v1.ConvertTransformTypeString, Format: v1.ConvertTransformFormatNone}: func(i any) (any, error) { //nolint:unparam // See note above.
-		return strconv.FormatFloat(i.(float64), 'f', -1, 64), nil
-	},
-	{From: v1.ConvertTransformTypeFloat64, To: v1.ConvertTransformTypeInt64, Format: v1.ConvertTransformFormatNone}: func(i any) (any, error) { //nolint:unparam // See note above.
-		return int64(i.(float64)), nil
-	},
-	{From: v1.ConvertTransformTypeFloat64, To: v1.ConvertTransformTypeBool, Format: v1.ConvertTransformFormatNone}: func(i any) (any, error) { //nolint:unparam // See note above.
-		return i.(float64) == float64(1), nil
-	},
-}
-
-// ResolveConvert resolves a Convert transform.
+// ResolveConvert resolves a Convert transform by looking up the appropriate
+// conversion function for the given input type and invoking it.
 func ResolveConvert(t v1.ConvertTransform, input any) (any, error) {
-	from := reflect.TypeOf(input).Kind().String()
-	if from == v1.ConvertTransformTypeInt {
-		from = v1.ConvertTransformTypeInt64
-	}
-	to := t.ToType
-	if to == v1.ConvertTransformTypeInt {
-		to = v1.ConvertTransformTypeInt64
-	}
-
-	format := v1.ConvertTransformFormatNone
-	if t.Format != nil {
-		format = *t.Format
-	}
-	switch format {
-	case v1.ConvertTransformFormatNone, v1.ConvertTransformFormatQuantity:
-		break
-	default:
-		return nil, errors.Errorf(errConvertFormatNotSupported, string(format))
-	}
-
-	switch from {
-	case to:
-		return input, nil
-	case v1.ConvertTransformTypeString, v1.ConvertTransformTypeBool, v1.ConvertTransformTypeInt64, v1.ConvertTransformTypeFloat64:
-		break
-	default:
-		return nil, errors.Errorf(errFmtConvertInputTypeNotSupported, reflect.TypeOf(input).Kind().String())
```

Comment:
Ok, I guess what happened is; we moved these checks out into `Validate` methods but after the discussion [here](https://github.com/crossplane/crossplane/pull/3921#discussion_r1150817325) we removed them completely 😅 Is this the case for other instances as well?

While I still believe we will not hit them at runtime due to Kube API Schema validation, I am hesitant to change the existing code as part of introducing admission webhook, and AFAIR, these runtime checks are prevalent in the codebase.

Reply:
Sounds good. I don't have a strong opinion as long as we keep the runtime functionality the same as before.